terratorch/datasets/pastis.py

```diff
@@ -0,0 +1,312 @@
+                    - the (height, width) size of each parcel,
+                    - the semantic label of each parcel,
+                    - the semantic label of each pixel.
+            folds (list, optional): List of ints specifying which of the 5 official
+                folds to load. By default (when None is specified), all folds are loaded.
+            class_mapping (dict, optional): A dictionary to define a mapping between the
+                default 18 class nomenclature and another class grouping. If not provided, 
+                the default class mapping is used.
+            transform (callable, optional): A transform to apply to the loaded data 
+                (images, dates, and masks). By default, no transformation is applied.
+            truncate_image (int, optional): Truncate the time dimension of the image to 
+                a specified number of timesteps. If None, no truncation is performed.
+            pad_image (int, optional): Pad the time dimension of the image to a specified 
+                number of timesteps. If None, no padding is applied.
+            sats (list): Defines the satellites to use. If you are using PASTIS-R, you
+                have access to Sentinel-2 imagery and Sentinel-1 observations in Ascending
+                and Descending orbits, respectively S2, S1A, and S1D. For example, use
+                sats=['S2', 'S1A'] for Sentinel-2 + Sentinel-1 ascending time series,
+                or sats=['S2', 'S1A', 'S1D'] to retrieve all time series. If you are using
+                PASTIS, only S2 observations are available.
+        """
+        if target not in ["semantic", "instance"]:
+            msg = f"Target '{target}' not recognized. Use 'semantic', or 'instance'."
+            raise ValueError(msg)
+        valid_sats = {"S2", "S1A", "S1D"}
+        for sat in sats:
+            if sat not in valid_sats:
+                msg = f"Satellite '{sat}' not recognized. Valid options are {valid_sats}."
+                raise ValueError(msg)
+
+        super().__init__()
+        self.data_root = data_root
+        self.norm = norm
+        self.reference_date = datetime(*map(int, reference_date.split("-")), tzinfo=timezone.utc)
+        self.class_mapping = (
+            np.vectorize(lambda x: class_mapping[x])
+            if class_mapping is not None
+            else class_mapping
+        )
+        self.target = target
+        self.sats = sats
+        self.transform = transform
+        self.truncate_image = truncate_image
+        self.pad_image = pad_image
+        # loads patches metadata
+        self.meta_patch = gpd.read_file(os.path.join(data_root, "metadata.geojson"))
+        self.meta_patch.index = self.meta_patch["ID_PATCH"].astype(int)
+        self.meta_patch.sort_index(inplace=True)
+
+        # stores table for each satalite date
+        self.date_tables = {s: None for s in sats}
+        self.date_range = np.array(range(-200, 600)) # date interval
```

Comment:
Any specific reason for this range? A comment to clarify might be good.

Reply:
This is the time range that was used in the paper. I will clarify that.